Added a function to find a device which starts with a given name

Baum55 opened the pull request:
Added a function to find a device which starts with a given name.

The Xilinx Vivado and SDK creates a device tree file which contains the base address as a part of the name (e.g. "axi_bram_ctrl@6fff0000").
It is not possible to implement a general search for the base adresse of a device with "three byname(const char *name)" when the base address have to be part of the search string.

Baum55 committed:
Added a function to find a device which starts with a given name

Added a function to find a device which starts with a given name.

The Xilinx Vivado and SDK creates a device tree file which contains the base address as a part of the name (e.g. "axi_bram_ctrl@6fff0000").
It is not possible to implement a general search for the base adresse of a device with "three byname(const char *name)" when the base address have to be part of the search string.

jviki commented on dtree.h:
```diff
@@ -137,6 +137,20 @@ struct dtree_dev_t *dtree_next(void);
+ * Returns NULL when not found or on error.
+ * On error sets error state.
+ */
+struct dtree_dev_t *dtree_startbyname(const char *name);
```
Please, just rename the function to something like _dtree_byname_prefix_ to stay consistent with original convention _bysomething_.

jviki reviewed (changes requested):
Thank you for contribution, please note the one simple comment and fix it. I'll approve the pull request then.

Baum55 committed:
function and parameter renamed to stay consistent

Renamed the function and parameter to stay consistent with the original convention:

Function: dtree_bysomething
Parameter: something

Baum55 commented:
Thanks for for the advice. It's definitely a better name for the function.

And thank you for the whole lib code.
